core/vm/instructions.go

```diff
@@ -756,23 +769,41 @@ func opCall(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory
 	value = math.U256(value)
 	// Get the arguments from the memory.
 	args := memory.GetPtr(inOffset.Int64(), inSize.Int64())
-
-	if value.Sign() != 0 {
-		gas += params.CallStipend
-	}
-	ret, returnGas, err := interpreter.evm.Call(contract, toAddr, args, gas, value)
-	if err != nil {
-		stack.push(interpreter.intPool.getZero())
-	} else {
+	methodId := args[0:4]
```

Comment:
I think this will blow up on parameter-less calls to fallback functions (where methodId is not set). Might need an `if` to protect against `args` being empty.

Reply:
Good catch! I'll add a check here

Reply:
I ended up [pulling `isCallTo` into its own function](https://github.com/op-optimism/go-ethereum/pull/1/commits/a61a624ad631a81f68020c633f80b94bbcbe391c)  and checking args length in there. Looks much cleaner now IMO ✨